src/utils/jss/renderInShadowRoot.tsx

```diff
@@ -25,7 +25,7 @@ export function renderInShadowRoot(node: React.ReactNode, shadow: ShadowRoot) {
     if (shadow.mode === 'open') console.warn('Do not render with open ShadowRoot!')
     ReactDOM.render(<RenderInShadowRootWrapper children={node} />, shadow as any)
     livingShadowRoots.add(shadow)
-    applyAdoptedStyleSheets()
+    applyAdoptedStyleSheet(shadow)
```

Comment:
If you don't refresh all the shadow root, there may be a CSS bug due to stylesheet inconsistency. (And I thought that already happened in the past)

Reply:
How did that bug happened? Every style changes invoke `adoptStylesheets` which do syncing stylesheet between all shadow roots. Now `applyAdoptedStyleSheet` appends a shadow root into the cached set `livingShadowRoots` immediately which ensure `adoptStylesheets` wouldn't be lost anyone.

Reply:
Wait for a while... I am working at this in hope of completely remove all the hacks that `adoptStylesheets` applies.

Reply:
@SunriseFox Those code have been revoked!